Fix returning translated title in solrsearch and listing endpoints.

njohner opened the pull request:
We fix returning the translated title in the `listing` and `solrsearch` endpoints. The check that we were making to decide whether to return a translated title was never passing, as `OGSolrContentListingObject` does not support `providedBy`.

For [CA-2649]

## Checklist
- [x] Changelog entry
- [x] Link to issue (Jira or GitHub) and backlink in issue (Jira)

[CA-2649]: https://4teamwork.atlassian.net/browse/CA-2649

tinagerber commented on opengever/api/solr_query_service.py:
```diff
@@ -102,9 +102,8 @@ def filename(obj):
-        attr = 'title_{}'.format(get_preferred_language_code())
-        return getattr(obj, attr, obj.Title())
+    if ITranslatedTitleSupport.__identifier__ in obj.object_provides:
+        return obj.getObject().Title(language=get_preferred_language_code())
```
Why do we now get the object when the title is translatable? That not good for performance

lukasgraf replied:
Good catch!

`title_de` and `title_fr` are not currently indexed in Solr, so they need to be fetched from the actual object. In the future we should index them, but for repository folders we'll need to take care to not index their raw values (`Umwelt`), but the composed ones prefixed with reference number (`7.3 Umwelt`).

Because objects with `ITranslatedTitle` are always structural, they will already be in the cache in most cases. Also, there's rather few of them compared to other objects, so I think the fallback to the object should be ok in this case.

njohner replied:
And to complement, we were already fetching the object before (there is no other way to get to the translated titles). It was just not obvious that we were fetching the object, because there is a fallback to getting attributes from the underlying object in `OpengeverCatalogContentListingObject`. I probably broke this with https://github.com/4teamwork/ftw.solr/pull/182, which tries to avoid that fallback whenever possible.

tinagerber commented on opengever/api/tests/test_listing.py:
```diff
@@ -185,6 +185,26 @@ class TestListingWithRealSolr(SolrIntegrationTestCase):
+        self.enable_languages()
+
+        self.login(self.regular_user, browser=browser)
+        view = '@listing?name=repository_folders&columns=title&columns=title'
```
```suggestion
        view = '@listing?name=repository_folders&columns:list=title'

lukasgraf commented on opengever/api/solr_query_service.py:
```diff
@@ -336,7 +335,8 @@ def hide_facet(self, facet):
                  additional_required_fields=['bumblebee_checksum', 'path']),
-    ListingField('title', 'Title', translated_title, 'sortable_title'),
+    ListingField('title', 'Title', translated_title, 'sortable_title',
+                 additional_required_fields=['object_provides']),
```
As discussed, I wouldn't always request `object_provides`. The list of provided interfaces is *huge* for most objects, and can easily make up 70-80% of a solr document if otherwise few fields were requests.

Instead, we should be able to do with check with a static list of portal types, and write a test that makes sure the list is up-to-date by looking up all FTIs and their behaviors.

njohner commented:
Ok I modified the implementation to use a static list of portal types that have the translated_title behavior and added a test to make sure the list is up-to-date.
Can someone give it another look?

lukasgraf reviewed (approved):
LGTM! 👍
